de.dlr.sc.virsat.fdir.core/src/de/dlr/sc/virsat/fdir/core/markov/algorithm/Bisimulation.java

```diff
@@ -0,0 +1,264 @@
+package de.dlr.sc.virsat.fdir.core.markov.algorithm;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+
+import org.eclipse.core.runtime.SubMonitor;
+
+import de.dlr.sc.virsat.fdir.core.markov.MarkovAutomaton;
+import de.dlr.sc.virsat.fdir.core.markov.MarkovState;
+import de.dlr.sc.virsat.fdir.core.markov.MarkovTransition;
+
+/**
+ * Class for performing Bisimulation
+ * 
+ * @author rama_vi
+ *
+ */
+public class Bisimulation {
+
+	protected MarkovAutomaton<MarkovState> ma;
+	protected Map<MarkovState, Set<MarkovState>> mapStateToBlock;
+
+	public Bisimulation(MarkovAutomaton<MarkovState> ma) {
+		this.ma = ma;
+
+	}
+
+	/**
+	 * Computes Equivalence Classes and quotient
+	 */
+	public void minimize() {
+		computeEquivalenceClasses();
+		computeQuotient();
+	}
+
+	/**
+	 * Assigns the states to respective blocks when they have similar labels
+	 * 
+	 * @return blocks
+	 */
+	protected Set<Set<MarkovState>> createInitialBlocks() {
+		Set<Set<MarkovState>> blocks = new HashSet<>();
+		mapStateToBlock = new HashMap<>();
+		for (MarkovState state : ma.getStates()) {
+			Set<MarkovState> block = getBlock(state, blocks);
+			if (!blocks.remove(block)) {
+				block = new HashSet<>();
+			}
+
+			block.add(state);
+			blocks.add(block);
+			mapStateToBlock.put(state, block);
+		}
+
+		return blocks;
+	}
+
+	/**
+	 * @param state
+	 * @param blocks
+	 * @return block The block to which a state belongs to
+	 */
+	private Set<MarkovState> getBlock(MarkovState state, Set<Set<MarkovState>> blocks) {
+		for (Set<MarkovState> block : blocks) {
+			if (belongsToBlock(block, state)) {
+				return block;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * checks whether a state belongs to block or not
+	 * 
+	 * @param block
+	 * @param state
+	 * @return boolean outputs true or false
+	 */
+	protected boolean belongsToBlock(Set<MarkovState> block, MarkovState state) {
+		List<Object> StateLabels = new ArrayList<Object>();
```

Comment:
Doesn't match variable name convention. Please check that all variable names are cameCase.